File: bungee/src/main/java/com/emanuelvini/nuven/core/bungee/BungeeMain.java
package com.emanuelvini.nuven.core.bungee;


import com.emanuelvini.nuven.core.bungee.configuration.database.DatabaseValue;
import com.emanuelvini.nuven.core.bungee.data.repository.ProfileRepository;
import com.emanuelvini.nuven.core.bungee.logger.BungeeLogger;
import com.emanuelvini.nuven.core.bungee.manager.TagManager;
import com.emanuelvini.nuven.core.bungee.registry.ConfigurationRegistry;
import com.emanuelvini.nuven.core.bungee.registry.PacketListenerRegistry;
import com.henryfabio.sqlprovider.connector.type.impl.MySQLDatabaseType;
import com.henryfabio.sqlprovider.executor.SQLExecutor;
import lombok.Getter;
import lombok.experimental.Accessors;
import lombok.val;
import net.md_5.bungee.api.ChatColor;
import net.md_5.bungee.api.ProxyServer;
import net.md_5.bungee.api.plugin.Plugin;

@Getter
public class BungeeMain extends Plugin {

    @Getter
    @Accessors(fluent = true)
    private static BungeeMain instance;

    private BungeeLogger logger;

    private TagManager tagManager;

    private ProfileRepository profileRepository;


    @Override
    public void onLoad() {
        super.onLoad();
        logger = new BungeeLogger(ProxyServer.getInstance().getConsole());

    }

    @Override
    public void onEnable() {
        super.onEnable();

        val configurationRegistry = new ConfigurationRegistry(this);
        configurationRegistry.register();

        try {

            tagManager = new TagManager();

            val executor = new SQLExecutor(
                    MySQLDatabaseType
                            .builder()
                            .database(DatabaseValue.get(DatabaseValue::database))
                            .username(DatabaseValue.get(DatabaseValue::username))
                            .password(DatabaseValue.get(DatabaseValue::password))
                            .address(DatabaseValue.get(DatabaseValue::address))
                            .build()
                            .connect()
            );
            profileRepository = new ProfileRepository(executor, tagManager);

            logger.success("Conectado a database com sucesso.");

            getProxy().registerChannel("nvcore:main");
            logger.success("Canais de comunicações registrados com sucesso.");

            val packetListenerRegistry = new PacketListenerRegistry(this);
            packetListenerRegistry.register();



            logger.success("Plugin habilitado com sucesso.");
        } catch (Exception e) {
            logger.error("Falha ao acessar o banco de dados, por favor informe abaixo:");
            e.printStackTrace();
        }



    }
}

File: bungee/src/main/kotlin/com/emanuelvini/nuven/core/bungee/BungeeMain.kt
package com.emanuelvini.nuven.core.bungee

import com.emanuelvini.nuven.core.bungee.data.repository.PreferencesRepository
import com.emanuelvini.nuven.core.bungee.data.repository.ProfileRepository
import com.emanuelvini.nuven.core.bungee.logger.BungeeLogger
import com.emanuelvini.nuven.core.bungee.manager.RoleManager
import com.emanuelvini.nuven.core.bungee.registry.primary.ConfigurationRegistry
import com.emanuelvini.nuven.core.bungee.registry.registries.CommandRegistry
import com.emanuelvini.nuven.core.bungee.registry.registries.PacketListenerRegistry
import com.emanuelvini.nuven.core.bungee.registry.registries.RoleRegistry
import com.emanuelvini.nuven.core.shared.database.DatabaseValue
import com.emanuelvini.nuven.core.shared.player.preferences.lobby.visibility.VisibilityPreference
import com.emanuelvini.nuven.core.shared.registry.Registry
import com.google.common.io.ByteStreams
import com.henryfabio.sqlprovider.connector.type.impl.MySQLDatabaseType
import com.henryfabio.sqlprovider.executor.SQLExecutor
import net.md_5.bungee.api.ProxyServer
import net.md_5.bungee.api.plugin.Plugin
import java.io.File
import java.util.concurrent.TimeUnit
import java.util.function.Consumer


class BungeeMain : Plugin() {
    val bungeeLogger: BungeeLogger = BungeeLogger(ProxyServer.getInstance().console)
    val settingsFolder: File = File(dataFolder, "settings")
    val roleManager: RoleManager = RoleManager()
    var profileRepository: ProfileRepository? = null
    var preferencesRepository: PreferencesRepository? = null
    var registries: List<Registry>? = null
    val preferences = listOf(
        VisibilityPreference()
    )
    override fun onLoad() {
        instance = this

    }

    override fun onEnable() {
        super.onEnable()

        val configurationRegistry = ConfigurationRegistry(this)
        configurationRegistry.register()
        try {
            val executor = SQLExecutor(
                MySQLDatabaseType
                    .builder()
                    .database(DatabaseValue[DatabaseValue::database])
                    .username(DatabaseValue[DatabaseValue::username])
                    .password(DatabaseValue[DatabaseValue::password])
                    .address(DatabaseValue[DatabaseValue::address])
                    .build()
                    .connect()
            )

            profileRepository =
                ProfileRepository(executor, roleManager)
            preferencesRepository = PreferencesRepository(executor, preferences)

            preferencesRepository!!.createTable()
            profileRepository!!.createTable()
            registries = listOf(
                PacketListenerRegistry(this),
                RoleRegistry(this, profileRepository!!),
                CommandRegistry(this)
            )
            registries!!.forEach(Consumer { obj: Registry -> obj.register() })
            bungeeLogger.success("Plugin habilitado com sucesso.")
        } catch (e: Exception) {
            bungeeLogger.error("Falha ao acessar o banco de dados, por favor informe abaixo:")
            e.printStackTrace()
        }
    }

    override fun onDisable() {
        super.onDisable()
        registries!!.forEach(Consumer { obj: Registry -> obj.unregister() })
        bungeeLogger.success("Plugin desabilitado com sucesso.")
    }

    companion object {
        var instance: BungeeMain? = null
    }
}
